opslevel/resource_opslevel_system.go

```diff
@@ -1,162 +1,224 @@
+		Name:        RequiredStringValue(system.Name),
+		Note:        OptionalStringValue(system.Note),
+		Owner:       OptionalStringValue(string(system.Owner.Id())),
+	}
+	return systemDataSourceModel, diags
+}
+
+func (r *SystemResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_system"
+}
+
+func (r *SystemResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "System Resource",
+
+		Attributes: map[string]schema.Attribute{
+			"aliases": schema.ListAttribute{
+				ElementType: types.StringType,
+				Description: "The aliases of the system.",
+				Computed:    true,
+			},
+			"description": schema.StringAttribute{
+				Description: "The description for the system.",
+				Optional:    true,
+			},
+			"domain": schema.StringAttribute{
+				Description: "The id of the parent domain this system is a child for.",
+				Optional:    true,
+				Validators:  []validator.String{IdStringValidator()},
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of the system.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name for the system.",
+				Required:    true,
+			},
+			"note": schema.StringAttribute{
+				Description: "Additional information about the system.",
+				Optional:    true,
+			},
+			"owner": schema.StringAttribute{
+				Description: "The id of the team that owns the system.",
+				Optional:    true,
+				Validators:  []validator.String{IdStringValidator()},
+			},
+		},
+	}
+}
+
+func (r *SystemResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel SystemResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	systemInput := opslevel.SystemInput{
+		Name:        planModel.Name.ValueStringPointer(),
+		Description: planModel.Description.ValueStringPointer(),
+		OwnerId:     opslevel.NewID(planModel.Owner.ValueString()),
+		Note:        planModel.Note.ValueStringPointer(),
+	}
+	if planModel.Domain.ValueString() != "" {
+		systemInput.Parent = opslevel.NewIdentifier(planModel.Domain.ValueString())
+	}
+	system, err := r.client.CreateSystem(systemInput)
+	if err != nil || system == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create system, got error: %s", err))
+		return
+	}
+	stateModel, diags := NewSystemResourceModel(ctx, *system)
+	if diags != nil && diags.HasError() {
+		resp.Diagnostics.Append(diags...)
```

Comment:
Should this be called before line 135? (does not need to be inside the `if`)

Reply:
Yeah, this can be tidied up a bit